src/compute/src/server.rs

```diff
@@ -185,14 +243,149 @@ impl<'w, A: Allocate> Worker<'w, A> {
+                            build_part.into_iter().zip(builds_parts.iter_mut())
+                        {
+                            objects_to_build.push(BuildDesc {
+                                id: build_desc.id,
+                                plan,
+                            });
+                        }
+                    }
+                    // Each list of build descriptions results in a dataflow description.
+                    for (dataflows_part, objects_to_build) in
+                        dataflows_parts.iter_mut().zip(builds_parts)
+                    {
+                        dataflows_part.push(DataflowDescription {
+                            source_imports: dataflow.source_imports.clone(),
+                            index_imports: dataflow.index_imports.clone(),
+                            objects_to_build,
+                            index_exports: dataflow.index_exports.clone(),
+                            sink_exports: dataflow.sink_exports.clone(),
+                            as_of: dataflow.as_of.clone(),
+                            debug_name: dataflow.debug_name.clone(),
+                            id: dataflow.id,
+                        });
+                    }
+                }
+                dataflows_parts
+                    .into_iter()
+                    .map(ComputeCommand::CreateDataflows)
+                    .collect()
+            }
+            command => vec![command; parts],
+        }
+    }
+
+    fn setup_channel_and_run_client(
+        &mut self,
+        command_rx: CommandReceiver,
+        response_tx: ResponseSender,
+    ) {
+        let cmd_queue = Rc::new(RefCell::new(
+            VecDeque::<Result<ComputeCommand, TryRecvError>>::new(),
+        ));
+        let peers = self.timely_worker.peers();
+        let idx = self.timely_worker.index();
+
+        {
+            let rx_activator = Arc::clone(&self.rx_activator);
+            let cmd_queue = Rc::clone(&cmd_queue);
+
+            self.timely_worker.dataflow::<u64, _, _>(move |scope| {
+                let src = source(scope, "Command source", |mut cap, info| {
+                    let activator = scope.sync_activator_for(&info.address[..]);
+                    {
+                        *rx_activator.lock().unwrap() = Some(activator);
+                    }
+
+                    move |output| {
+                        //println!("w{:?} Calling non-blocking recv", &idx);
+                        if idx == 0 {
+                            // We turn the TryRecvError into an Option type, as it doesnt
+                            // implemenet the right traits to fulfill Timelys ExchangeData
+                            //
+                            match command_rx.try_recv() {
+                                Ok(cmd) => {
+                                    // get some data and send it.
+                                    let time = cap.time().clone();
```

Comment:
This will assign a new timestamp to every command, which could be a bit overwhelming if there are many small commands received. I think a better approach is to collect all data available at the current time (i.e., collect the output from `try_recv` in a loop, and send all that data at the same timestamp.)